server/store/store_test.go

```diff
@@ -0,0 +1,148 @@
+	store.api = api
+	return store, api
+}
+
+func TestGetAvatarCache(t *testing.T) {
+	for _, test := range []struct {
+		Name                 string
+		SetupAPI             func(*plugintest.API)
+		ExpectedErrorMessage string
+	}{
+		{
+			Name: "GetAvatarCache: Error while getting the avatar cache",
+			SetupAPI: func(api *plugintest.API) {
+				api.On("KVGet", avatarKey+testutils.GetID()).Return(nil, testutils.GetInternalServerAppError("unable to get the avatar cache"))
+			},
+			ExpectedErrorMessage: "unable to get the avatar cache",
+		},
+		{
+			Name: "GetAvatarCache: Valid",
+			SetupAPI: func(api *plugintest.API) {
+				api.On("KVGet", avatarKey+testutils.GetID()).Return([]byte("mock data"), nil)
+			},
+		},
+	} {
+		t.Run(test.Name, func(t *testing.T) {
+			assert := assert.New(t)
+			store, api := setupTestStore(&plugintest.API{})
+			test.SetupAPI(api)
+			resp, err := store.GetAvatarCache(testutils.GetID())
+
+			if test.ExpectedErrorMessage != "" {
+				assert.Contains(err.Error(), test.ExpectedErrorMessage)
+				assert.Nil(resp)
+			} else {
+				assert.Nil(err)
+				assert.NotNil(resp)
+			}
+		})
+	}
+}
+
+func TestSetAvatarCache(t *testing.T) {
+	for _, test := range []struct {
+		Name                 string
+		SetupAPI             func(*plugintest.API)
+		ExpectedErrorMessage string
+	}{
+		{
+			Name: "SetAvatarCache: Error while setting the avatar cache",
+			SetupAPI: func(api *plugintest.API) {
+				api.On("KVSetWithExpiry", avatarKey+testutils.GetID(), []byte{10}, int64(avatarCacheTime)).Return(testutils.GetInternalServerAppError("unable to set the avatar cache"))
+			},
+			ExpectedErrorMessage: "unable to set the avatar cache",
+		},
+		{
+			Name: "SetAvatarCache: Valid",
+			SetupAPI: func(api *plugintest.API) {
+				api.On("KVSetWithExpiry", avatarKey+testutils.GetID(), []byte{10}, int64(avatarCacheTime)).Return(nil)
+			},
+		},
+	} {
+		t.Run(test.Name, func(t *testing.T) {
+			assert := assert.New(t)
+			store, api := setupTestStore(&plugintest.API{})
+			test.SetupAPI(api)
+			err := store.SetAvatarCache(testutils.GetID(), []byte{10})
+
+			if test.ExpectedErrorMessage != "" {
+				assert.Contains(err.Error(), test.ExpectedErrorMessage)
+			} else {
+				assert.Nil(err)
+			}
+		})
+	}
+}
+
+func TestCheckEnabledTeamByTeamID(t *testing.T) {
+	for _, test := range []struct {
+		Name           string
+		SetupAPI       func(*plugintest.API)
+		EnabledTeams   func() []string
+		ExpectedResult bool
+	}{
+		{
+			Name:     "CheckEnabledTeamByTeamID: Emmpty enabled team",
+			SetupAPI: func(api *plugintest.API) {},
+			EnabledTeams: func() []string {
+				return []string{""}
+			},
+			ExpectedResult: true,
+		},
+		{
+			Name: "CheckEnabledTeamByTeamID: Unable to get the team",
+			SetupAPI: func(api *plugintest.API) {
+				api.On("GetTeam", "mockTeamID").Return(nil, testutils.GetInternalServerAppError("unable to get the team"))
+			},
+			EnabledTeams: func() []string {
+				return []string{"mockTeamsTeam"}
+			},
```

Comment:
I would prefer to see here the `ExpecteResult` field, because even if that is working because the zero value of `bool` type, I think makes sense to clearly show the expectation there, because the expectation in this case has a clear meaning.

Reply:
And of course the same for all the test cases here.